plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/create-simple-product-block-editor.spec.js

```diff
@@ -46,182 +58,420 @@ test.describe( 'General tab', { tag: '@gutenberg' }, () => {
-				.locator( '[data-title="Paragraph"]' )
-				.last()
-				.fill( productData.descriptionParagraph );
+				await descriptionSimpleParagraph.fill(
+					productData.descriptionSimple
+				);
+			} );
 
-			await page.getByRole( 'button', { name: 'Done' } ).click();
-
-			const previewContainerIframe = page
-				.locator( '.block-editor-block-preview__container' )
-				.frameLocator( 'iframe[title="Editor canvas"]' );
-
-			const descriptionTitle = previewContainerIframe.locator(
-				'[data-title="Heading"]'
-			);
-			const descriptionInitialParagraph = previewContainerIframe
-				.locator( '[data-title="Paragraph"]' )
-				.first();
-			const descriptionSecondParagraph = previewContainerIframe
-				.locator( '[data-title="Paragraph"]' )
-				.last();
-
-			await expect( descriptionTitle ).toHaveText(
-				productData.descriptionTitle
-			);
-			await expect( descriptionInitialParagraph ).toHaveText(
-				productData.descriptionSimple
-			);
-			await expect( descriptionSecondParagraph ).toHaveText(
-				productData.descriptionParagraph
-			);
-
-			await descriptionTitle.click();
-
-			await expect(
-				page.getByText( 'Edit in full editor' )
-			).toBeVisible();
-
-			await page
-				.locator(
-					'[data-template-block-id="basic-details"] .components-summary-control'
-				)
-				.last()
-				.fill( productData.summary );
-
-			// We blur the summary field to hide the toolbar before clicking on the regular price field.
-			await page
-				.locator(
-					'[data-template-block-id="basic-details"] .components-summary-control'
-				)
-				.last()
-				.blur();
-
-			const regularPrice = page
-				.locator( 'input[name="regular_price"]' )
-				.first();
-			await regularPrice.waitFor( { state: 'visible' } );
-			await regularPrice.click();
-			await regularPrice.fill( productData.productPrice );
-
-			const salePrice = page
-				.locator( 'input[name="sale_price"]' )
-				.first();
-			await salePrice.waitFor( { state: 'visible' } );
-			await salePrice.click();
-			await salePrice.fill( productData.salePrice );
-
-			await page
-				.locator( '.woocommerce-product-header__actions' )
-				.getByRole( 'button', {
-					name: 'Publish',
-				} )
-				.click();
-
-			await expect(
-				page.getByLabel( 'Dismiss this notice' )
-			).toContainText( 'Product published' );
-
-			const title = page.locator( '.woocommerce-product-header__title' );
-
-			// Save product ID
-			const productIdRegex = /product%2F(\d+)/;
-			const url = page.url();
-			const productIdMatch = productIdRegex.exec( url );
-			productId = productIdMatch ? productIdMatch[ 1 ] : null;
-
-			await expect( productId ).toBeDefined();
-			await expect( title ).toHaveText( productData.name );
+			await test.step( 'add full product description', async () => {
+				const descriptionSimpleParagraph = page.locator(
+					'[data-template-block-id="product-description__content"] > p'
+				);
+
+				// Helps to ensure that block toolbar appears
+				await descriptionSimpleParagraph.click();
```

Comment:
Hmm I wonder if there is a more reliable way to do this?
I get a feeling this may cause some flaky test behaviour, I tried this locally and it would fail sometimes as the toolbar didn't show up.
Maybe hitting `Escape` would be more reliable: `await page.keyboard.press( 'Escape' );`

It looks like the toolbar disappears when the editor thinks its typing, it looks like the logic for this is handled here: https://github.com/WordPress/gutenberg/blob/a5c6aed30ec7913876f8f7ed239665f110cb8d50/packages/block-editor/src/components/observe-typing/index.js#L55

There are a couple ways to tell it to `stopTyping`:
- When the mouse moves
- On focus
- When the user hits escape or tab ( although tab does not work for me )
- Or selecting text using your keyboard for example

Reply:
I like that... I'll give it a try. Anything to avoid flaky tests.